### PR TITLE
Split releases into 3 workflows.

### PR DESCRIPTION
**Pull request by timcosgrove**
# Description
Currently prod staging and dev releases are handled by one workflow. This PR decouples those, in particular so that prod content release is not dependent on staging and dev timing.

It also changes dev and staging to run once nightly, and to be triggered by successful merges to the main branch.

## Ticket
Relates to #714 

## Developer Task

```[tasklist]
- [x] PR submitted against the `main` branch of `next-build`.
- [x] Link to the issue that this PR addresses (if applicable).
- [x] Define all changes in your PR and note any changes that could potentially be breaking changes.
- [x] PR includes steps to test your changes and links to these changes in the Tugboat preview (if applicable).
- [x] Provided before and after screenshots of your changes (if applicable).
- [x] Alerted the #accelerated-publishing Slack channel to request a PR review.
- [x] You understand that once approved, you are responsible for merging your changes into `main`. (Note that changes to `main` will move automatically into production.)
```

## Testing
Because these are effectively new workflows, they cannot be tested without being merged. I recommend
- we review from a code perspective
- merge and test
- immediately revert out if there are issues that cannot be resolved quickly

Testing should be checking each of the following:
- each individual workflow can be triggered manually
- prod workflow is running as a cron workflow per the schedule
- dev and staging workflows run as cron workflow per their schedules
- dev and staging workflows are triggered when new code is merged to main and is successfully tested and a new production tag is created

### CODE REVIEW

File: .github/workflows/content-release.yml [hunk 1]
```diff
@@ -5,19 +5,18 @@ permissions:
   contents: read  # This is required for actions/checkout
 
 on:
+  # Runs on API call. Used for CMS-driven build triggers.
   repository_dispatch:
     types: [content-release]
-  workflow_run:
```
Comment (timcosgrove) [inline .github/workflows/content-release.yml:10]: This trigger off production tag is deliberately removed here. We add it to the dev and staging builds.

File: .github/workflows/recurring-release-dev.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: "Recurring release: Dev"
+on:
+  # Can be manually triggered
+  workflow_dispatch:
+  # Runs 5:05 am EDT Monday to Friday.
```
Comment (timcosgrove) [inline .github/workflows/recurring-release-dev.yml:5]: Changed to run once nightly. The timing is 5:05 am to allow any rebuild of the Tugboat content source to be complete before trying to run.

Comment (timcosgrove) [reply .github/workflows/recurring-release-dev.yml:5]: 5:35 am for staging; see below.

File: .github/workflows/recurring-release-prod.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+name: "Recurring release: Prod"
+on:
+  # Can be manually triggered
+  workflow_dispatch:
+  # Runs automatically every 30 minutes from 8am to 8pm Monday to Friday.
+  # This currently UTC -> EDT.
+  schedule:
+    - cron: "*/30 0,12-23 * * 1-5"
```
Comment (timcosgrove) [inline .github/workflows/recurring-release-prod.yml:8]: This is modified from the original timeframe to shift it to running 8am to 8pm ET.

File: .github/workflows/content-release.yml [hunk 4]
```diff
@@ -5,19 +5,15 @@ permissions:
   contents: read  # This is required for actions/checkout
 
 on:
-  repository_dispatch:
```
Comment (timcosgrove) [inline .github/workflows/content-release.yml:8]: This is deliberately removed here and moved to the prod release.

File: .github/workflows/content-release-prod.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+name: "Content release: Prod"
+on:
+  # Can be manually triggered
+  workflow_dispatch:
```
Comment (nfpappas-oddball) [inline .github/workflows/content-release-prod.yml:4]: If we go to production, we probably probably don't want to be able to trigger a build manually, maybe. Wanted to call that concern out there.

File: .github/workflows/content-release-staging.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  workflow_run:
+    workflows: ['Create Production Tag']
+    types: [completed]
+    branches: [main]
+
+jobs:
+  content-release-staging:
+    # This job should run for any valid event besides workflow_run, or workflow_run if the conclusion was successful.
+    if: ${{ github.event_name != 'workflow_run' || github.event.workflow_run.conclusion == 'success' }}
+    uses: department-of-veterans-affairs/next-build/.github/workflows/content-release.yml@main
```
Comment (nfpappas-oddball) [inline .github/workflows/content-release-staging.yml:19]: Love the code re-use!

**Review by nfpappas-oddball — APPROVED** [review]
May need to consider the ramifications of being able to run production release on demand once our code is actually in prod. Otherwise Love the code re-use and find it to be an overall better solution then we had before. RIP recurring-release.

File: .github/workflows/content-release-prod.yml (hunk 5 above)
Comment (timcosgrove) [reply .github/workflows/content-release-prod.yml:4]: We have times where we need to fire Content Build manually if the normal triggers are failing. I think it's ok.

However, that brings up a point I missed: these workflows need concurrency management. I'll add that in.

**Review by nfpappas-oddball — APPROVED** [review]

**Pull request merged**